src/compiler/crystal/macros/methods.cr

```diff
@@ -142,6 +144,20 @@ module Crystal
       raise SkipMacroException.new(@str.to_s)
     end
 
+    def interpret_read_file(node)
+      if node.args.size == 1
+        node.args[0].accept self
+        filename = @last.to_macro_id
+        if File.exists?(filename)
```

Comment:
I think the full filename's logic should be enhanced. Check how the `run` macro method is implemented.

Reply:
Done and ready for re-review :)